### PR TITLE
Feature/equal merit tie breakers

### PR DESCRIPTION
**Pull request by HalcyonJAC**
---
PREVIEW:DEVELOP

### CODE REVIEW

File: src/views/Exercises/Tasks/QualifyingTests/QualifyingTest/Edit.vue [hunk 1]
```diff
@@ -92,8 +96,8 @@ export default {
 
     const defaults = {
       title: null,
-      startDate: this.getTimelineDate(exercise, data.type, 'start') || null,
-      endDate: this.getTimelineDate(exercise, data.type, 'end') || null,
+      startDate: (data.isTieBreaker ? this.getEMZDate(exercise, 'Start') : this.getTimelineDate(exercise, data.type, 'Start')) || null,
+      endDate: (data.isTieBreaker ? this.getEMZDate(exercise, 'End') : this.getTimelineDate(exercise, data.type, 'StarEndt')) || null,
```
Comment (mbrookeswebdev) [inline src/views/Exercises/Tasks/QualifyingTests/QualifyingTest/Edit.vue:100]: I think you meant to put 'End' here?

Comment (HalcyonJAC) [reply src/views/Exercises/Tasks/QualifyingTests/QualifyingTest/Edit.vue:100]: Yep, it definately isn't going to find `StarEndt`. Well spotted @mbrookeswebdev

Comment (HalcyonJAC) [reply src/views/Exercises/Tasks/QualifyingTests/QualifyingTest/Edit.vue]: now fixed :)

**Review by warrensearle — APPROVED** [review]
Looks good to me 👍

**Review by mbrookeswebdev — APPROVED** [review]

File: src/views/Exercises/Show/Reports/QualifyingTestReports/QualifyingTestReport/ViewScore.vue [hunk 2]
```diff
@@ -2,7 +2,7 @@
   <div>
     <div class="govuk-grid-column-full govuk-!-margin-bottom-1">
       <h2 class="govuk-heading-m">
-        Qualifying Test Report
+        {{ tieBreakers ? 'Equal Merit Tie-breaker' : 'Qualifying Test' }} Report
```
Comment (tomlovesgithub) [inline src/views/Exercises/Show/Reports/QualifyingTestReports/QualifyingTestReport/ViewScore.vue:5]: is there someway we extract this ternary to a filter? (à la `formatDate`)

File: src/views/Exercises/Tasks/QualifyingTests/QualifyingTest.vue [hunk 3]
```diff
@@ -54,6 +54,8 @@ export default {
     },
     redirectToPage() {
       // this.$router.replace({ name: 'page-not-found' });
+      // TODO: If the requested test is an Equal Merit tie-breaker redirects to the EM tie-breakers page
```
Comment (tomlovesgithub) [inline src/views/Exercises/Tasks/QualifyingTests/QualifyingTest.vue:57]: is this still todo?

**Review by tomlovesgithub — APPROVED** [review]

**Comment by KateMJAC** [comment]
@HalcyonJAC, sorry to ask again, but can you please regenerate the PR preview URL?

**Comment by KateMJAC** [comment]
@HalcyonJAC I tested this PR and had one intial suggestion and then ran into an issue early on. 

1) teams will have a dry run for their EMZ tie-break, but they will also need to run a main test. We should have some kind of differentiation between the main test and a dry run. We need an option for people to select the scenario test (main test) and the dry run: 



2. I was told that I couldn't create the test becuase a QT was open for the exercise, but I don't think there was. 

I thought this issue could be because there was QTs open across the develop app, as we had discussed not allowing QTs and EMZ test to run at the same time????

I can retest tomorrow as well, just in case.

**Comment by HalcyonJAC** [comment]
[ ] Update the 'Create an EM Tie-breaker test' page so that Scenario Test is visible and chosen automatically, so that it is clear to the user that the test being created is a scenario test. Note: The user will still be able to choose whether it is a dry run or not.

**Review by KateMJAC — COMMENTED** [review]
Sent to UTG for feedback. Update to follow.

**Comment by KateMJAC** [comment]
Sent to the UTG a second time, after last week's testers did not complete.

**Review by KateMJAC — COMMENTED** [review]
Ticket not tested by prevous week's testers. 

Sent to UTG on 29/06/21. Awaiting feedback.

**Review by KateMJAC — COMMENTED** [review]
1 positive test confirmed. Sent to w/c 5 July testers for more feedback.

**Review by KateMJAC — COMMENTED** [review]
Unable to release this work due to merge conflicts. Awaiting action by devs.

**Comment by warrensearle** [comment]
Have fixed the merge conflicts however there are a few fixes still to make

**Comment by londonprojects** [comment]
@warrensearle @tomlovesgithub Is there anything needed from this ticket?

**Comment by warrensearle** [comment]
Hey @londonprojects, @tomlovesgithub and I are pairing today to sort out these merge conflicts

**Comment by joy-ade** [comment]
@Halcyon can you regenerate the PR pls.

**Comment by HalcyonJAC** [comment]
Merge conflicts resolved and router bugs fixed. PR url regenerated. 

It would be good to get a re-review of this and sent for testing again.

**Comment by mbrookeswebdev** [comment]
@HalcyonJAC when I click on 'Equal merit tie-breakers' link it takes me back to Exercises list -

EMP1
EMP2

**Comment by HalcyonJAC** [comment]
Thanks @mbrookeswebdev it looks like the routes were merged into the wrong place after staged applications maybe? I've edited and it appears to work now. Please could you try again?

**Comment by mbrookeswebdev** [comment]
I don't think I am doing anything wrong but the Create tests button is disabled -

EMP

**Comment by joy-ade** [comment]
@HalcyonJAC @JeremykJAC Thanks for resolving the merged conflict however I got this error message while l was creating a Equal Merit Tie broke test. I know Kate flagged  this issues up sometime ago. Can we look this pls. I should be able to input and date without having to change the date in my timeline. The scenario for qualifying test should be mirrored into Equal Merit Tie broke test.

[Capture.PNG99.PNG](https://images.zenhubusercontent.com/5f7b3a91edd7a0a890566450/6a78deec-85d4-46f0-ae09-f727e0fb26cd)

**Review by joy-ade — APPROVED** [review]
Approved by UTG

**Review by warrensearle — APPROVED** [review]
Code changes look good to me 👍

**Comment by JeremykJAC** [comment]
Hi @joy-ade 

The reason this is happening is because there is an EMT second stage date set for the 22/07/21 and therefore the system requires the first EMT to be completed. 

This was included on the development plan:

- [ ] 'When creating a EM tie-breaker the start and end dates for the test should default to the Equal Merit Second Phase start and end dates for the exercise'.

Would you like us to ignore the second stage and allow the end date to be set afterwards?

**Comment by joy-ade** [comment]
@KateMJAC FYI

JeremykJAC
commented 20 minutes ago
Hi @joy-ade

The reason this is happening is because there is an EMT second stage date set for the 22/07/21 and therefore the system requires the first EMT to be completed.

This was included on the development plan:

 'When creating a EM tie-breaker the start and end dates for the test should default to the Equal Merit Second Phase start and end dates for the exercise'.
Would you like us to ignore the second stage and allow the end date to be set afterwards?

**Comment by KateMJAC** [comment]
@joy-ade, yesterday I outlined to Jeremy that there may be an outlier where a candidate needs to sit the test at a different date, after the intial test.

**Comment by JeremykJAC** [comment]
@KateMJAC We will remove the constraint that stops the tie breaker end date being forced to be before the second stage date. If this is ok, then we will do this first thing and hopefully we can get this ticket moving again :)

**Comment by HalcyonJAC** [comment]
I have done that now. Darrel.

**Review by KateMJAC — APPROVED** [review]

**Pull request merged**